### PR TITLE
feature/shared file systems: list shares

### PR DESCRIPTION
**Pull request by mvalkon**
This pull request implements [`listing of shares`](http://developer.openstack.org/api-ref/shared-file-systems/?expanded=list-shares-detail#list-shares) with and without details

Manila implementation details:
https://github.com/openstack/manila/blob/16c522bf5e5c3d37018caef11dd7852462193327/manila/api/v1/shares.py#L110
https://github.com/openstack/manila/blob/16c522bf5e5c3d37018caef11dd7852462193327/manila/api/v1/shares.py#L153

For #129

### CODE REVIEW

**Comment by mvalkon** [comment]
The [api docs][1] are describing both `query` and `body` parameters for listing shares, however I could only find the `query` implementation in manila source. If anyone knows more about the details, feel free to comment.

[1]: http://developer.openstack.org/api-ref/shared-file-systems/?expanded=list-shares-detail#list-shares

File: openstack/sharedfilesystems/v2/shares/results.go [hunk 1]
```diff
@@ -2,12 +2,13 @@ package shares
...
 import (
 	"github.com/gophercloud/gophercloud"
+	"github.com/gophercloud/gophercloud/pagination"
 )
 
 // Share contains all information associated with an OpenStack Share
 type Share struct {
 	// The availability zone of the share
-	AvailabilityZone string `json:"availability_zone"`
+	AvailabilityZone string `json:"availability_zone,omitempty"`
```
Comment (mvalkon) [inline openstack/sharedfilesystems/v2/shares/results.go]: Not sure if this is a good idea to have each of the fields as `omitempty`. It allows us to re-use the type for short and long listing though. Open to suggestions on better alternatives

File: openstack/sharedfilesystems/v2/shares/results.go [hunk 2]
```diff
@@ -76,6 +77,20 @@ func (r commonResult) Extract() (*Share, error) {
 	return s.Share, err
 }
 
+// SharePage is a pagination.pager that is returned from a call to the List func
+type SharePage struct {
+	pagination.SinglePageBase
```
Comment (mvalkon) [inline openstack/sharedfilesystems/v2/shares/results.go]: Based on the discussion in #134 I'm starting to question if `SingleBasePage` should be a `MarkerBasePage`. Given that in the Manila API we support `limit` and `offset`, maybe it should be? 

I have a naive acceptance test for pagination calling `EachPage()` for shares, which seems to work with `SinglePageBase`, but I'm not confident that the test would reveal a problem.

**Comment by mvalkon** [comment]
@jrperritt do you have any comments regarding this one while #128 is waiting for a review?

**Comment by mvalkon** [comment]
@jrperritt this one should be ready for review.

**Comment by jrperritt** [comment]
Maybe I'm missing something, but I only see 2 files changed, and they're acceptance tests. Nothing wrong with that, but are the title and description correct?

**Comment by mvalkon** [comment]
@jrperritt you're correct. I've apparently pushed the wrong local branch. I'll have to rebase the correct one, re-run the tests and push again. I'll tag this as WIP until I'm sure it works.

**Comment by mvalkon** [comment]
@jrperritt finally managed to setup acceptance test environment and re-ran tests. Everything passes, please review.

File: openstack/sharedfilesystems/v2/shares/requests.go [hunk 3]
```diff
@@ -79,3 +80,67 @@ func Get(client *gophercloud.ServiceClient, id string) (r GetResult) {
...
+}
+
+// ListOpts is used to filter the List() call
+type ListOpts struct {
+	// Limit can be used to limit the query results
+	Limit int `q:"limit"`
+	// Offset can be used to define the starting point of share listing
+	Offset int `q:"offset"`
+	// Admin-only option to list shares from all tenants
+	AllTenants bool `q:"tenant_id"`
```
Comment (jtopjian) [inline openstack/sharedfilesystems/v2/shares/requests.go]: Should this be `q:"all_tenants"` ?

Comment (mvalkon) [inline openstack/sharedfilesystems/v2/shares/requests.go]: Most likely should, I will re-check this PR asap.

**Comment by jtopjian** [comment]
This was implemented in #1095

**Pull request closed without merging**